lib/cache/ResolverCachePlugin.js

```diff
@@ -218,8 +242,14 @@ class ResolverCachePlugin {
 								if (request._ResolverCachePluginCacheMiss || !fileSystemInfo) {
 									return callback();
 								}
-								const identifier = `/resolve/${type}${optionsIdent}${objectToString(
-									request,
+								const r = { ...request };
+
+								if (r.request) {
+									r.request = absolutify(r.path, r.request, this);
```

Comment:
That's a bit weird. And also pretty problematic from performance standpoint.

request should stay unmodified. `write/readPath` should not do destructive modifications to it. If it is already relative before serialization, it should stay relative after modification. Seem like it works this way, so not sure why this is needed.